### ISSUE TITLE
`TypeError: 'Dist' object is not subscriptable` on simple script

### ISSUE THREAD

**Issue by tadeu** [conda-tools/conda-execute#41]
I'm trying to execute `conda execute -f -v small_test.py` on Windows, on this simple script:

```python
#!/usr/bin/env conda-execute
"""
Simple test
"""

# conda execute
# env:
#  - python >=3
# run_with: python

print('Hello, world!')
```

and I'm getting this output:

```
Arguments passed: Namespace(code=None, force_env=True, path='small_test.py', quiet=False, remaining_args=[], verbose=True)
DEBUG conda-execute:main(173): Arguments passed: Namespace(code=None, force_env=True, path='small_test.py', quiet=False, remaining_args=[], verbose=True)
Using specification:
env: [python >=3]
run_with: [python]

INFO conda-execute:execute(87): Using specification:
env: [python >=3]
run_with: [python]

Fetching package metadata ...
Solving package specifications:


INFO conda-tmpenv:create_env(75):

Traceback (most recent call last):
  File "C:\Miniconda3\Scripts\conda-execute-script.py", line 11, in <module>
    load_entry_point('conda-execute==0.8.0', 'console_scripts', 'conda-execute')()
  File "C:\Miniconda3\lib\site-packages\conda_execute\execute.py", line 197, in main
    exit(execute(path, force_env=args.force_env, arguments=args.remaining_args))
  File "C:\Miniconda3\lib\site-packages\conda_execute\execute.py", line 89, in execute
    env_prefix = create_env(env_spec, force_env, spec.get('channels', []))
  File "C:\Miniconda3\lib\site-packages\conda_execute\tmpenv.py", line 79, in create_env
    dist_name = tar_name[:-len('.tar.bz2')]
TypeError: 'Dist' object is not subscriptable
```

This is my `conda info`:

```
Current conda install:

               platform : win-64
          conda version : 4.3.7
       conda is private : False
      conda-env version : 4.3.7
    conda-build version : 2.0.7
         python version : 3.5.2.final.0
       requests version : 2.12.4
       root environment : C:\Miniconda3  (writable)
    default environment : C:\Miniconda3
       envs directories : C:\Miniconda3\envs
          package cache : C:\Miniconda3\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : C:\Miniconda3\.condarc
           offline mode : False
             user-agent : conda/4.3.7 requests/2.12.4 CPython/3.5.2 Windows/10 Windows/10.0.14393
```

I'm on the root environmnet, and this is my `conda list`:

```
λ conda list
# packages in environment at C:\Miniconda3:
#
anaconda-client           1.5.2                    py35_0
cffi                      1.9.1                    py35_0
clyent                    1.2.2                    py35_0
conda                     4.3.7                    py35_0
conda-build               2.0.7                    py35_0
conda-build-all           1.0.1                    py35_1    conda-forge
conda-env                 2.6.0                         0    conda-forge
conda-execute             0.8.0                    py35_0    conda-forge
conda-smithy              2.0.1                    py35_0    conda-forge
console_shortcut          0.1.1                    py35_1
cryptography              1.7.1                    py35_0
filelock                  2.0.6                    py35_0
gitdb                     0.6.4                    py35_0    conda-forge
gitpython                 2.0.8                    py35_1    conda-forge
idna                      2.2                      py35_0
jinja2                    2.8                      py35_1
markupsafe                0.23                     py35_2
menuinst                  1.4.1                    py35_0
mock                      2.0.0                    py35_0    conda-forge
openssl                   1.0.2j                   vc14_0  [vc14]
pbr                       1.10.0                   py35_0    conda-forge
pip                       8.1.2                    py35_0
pkginfo                   1.3.2                    py35_0
psutil                    4.3.0                    py35_0    conda-forge
pyasn1                    0.1.9                    py35_0
pycosat                   0.6.1                    py35_1
pycparser                 2.17                     py35_0
pycrypto                  2.6.1                    py35_4
pygithub                  1.29                     py35_0    conda-forge
pyopenssl                 16.2.0                   py35_0
python                    3.5.2                         0
python-dateutil           2.5.3                    py35_0
pytz                      2016.7                   py35_0
pyyaml                    3.11                     py35_4
requests                  2.12.4                   py35_0
ruamel.yaml               0.11.11                  py35_1    conda-forge
ruamel_yaml               0.11.14                  py35_0
setuptools                23.0.0                   py35_0
six                       1.10.0                   py35_0
smmap                     0.9.0                    py35_0    conda-forge
vs2015_runtime            14.0.25123                    0
wheel                     0.29.0                   py35_0
```

**Comment by darkvertex** [comment]
I am erroring in the same line, except mine says:
```
TypeError: 'Dist' object has no attribute '__getitem__'
```

**Comment by jakirkham** [comment]
Basically this is an incompatibility of `conda-execute` with `conda` 4.3. One could downgrade to `conda` 4.2 to fix the issue.

As far as fixing this goes, one can call `Dist.from_string` to get a string to work with. In fact, this [function]( https://github.com/conda-forge/conda-build-all/blob/v1.0.4/conda_build_all/version_matrix.py#L268 ) would probably be worth taking to handle compatibility between `conda` 4.2 and 4.3.

Also as an FYI for @pelson, conda-forge is now running into this as well in various maintenance scripts as `conda` 4.3 has been released.

**Comment by pelson** [comment]
Ok, thanks @jakirkham. Agreed, the quick solution is downgrading your version of conda.
As a status update, the PR to make conda-execute compatible with conda 4.3 is in a healthy place, and I'd expect us to be able to release it in the next few days.

**Comment by jakirkham** [comment]
Thanks Phil.

**Comment by darkvertex** [comment]
Awesome! Thanks Phil.

**Comment by jakirkham** [comment]
This should be solved by v0.8.1. If upgrading does not fix the issue, please let us know.
